### PR TITLE
Java 11+28:  Remove now–non-applicable DSL code.  

### PR DESCRIPTION
**Pull request by RandomDSdevel**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.  
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.  (See below.)  
- [x] The commit message includes the cask’s name and version.  
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

~~Additionally, if **adding a new cask**:~~  

- [ ] ~~Named the cask according to the [token reference].~~  
- [ ] ~~`brew cask install {{cask_file}}` worked successfully.~~  
- [ ] ~~`brew cask uninstall {{cask_file}}` worked successfully.~~  
- [ ] ~~Checked there are no [open pull requests] for the same cask.~~  
- [ ] ~~Checked the cask was not [already refused].~~  
- [ ] ~~Checked the cask is submitted to [the correct repo].~~  

(N/A; not submitting a new cask.)  

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

---

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(Prompted by [this Discourse thread](https://discourse.brew.sh/t/homebrew-cask-java-packaging-puzzlement/3085/2).)  I based these changes on what's left in the copy of `/Library/Java/JavaVirtualMachines/jdk-11.jdk/Contents/Info.plist` distributed by upstream and as installed outside of Homebrew Cask, but let me know if any of these changes are either too aggressive or too conservative.  As for the newly-introduced `brew cask style` violations, I'm not sure how they popped out of me just deleting some surrounding code, but please suggest a fix for them.

### CODE REVIEW

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Note that I'm keeping this PR's staging branch rebased against `master` locally, but won't force-push-with-lease (which I prefer over plain old brute-force force-push) unless that is explicitly requested so as to avoid spamming CI.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'Bump' @commitay, as you appear to be the maintainer with the most contributions to the Java cask relatively recently.

File: Casks/java.rb [hunk 1]
```diff
@@ -29,13 +29,15 @@
...
+  uninstall pkgutil: "com.oracle.jdk-#{version.before_comma}",
+            delete:  [
+                       "/Library/Java/JavaVirtualMachines/jdk-#{version.before_comma}.jdk/Contents",
+                       '/Library/Java/Home',
+                       '/Library/Java/MacOS',
+                     ],
+            rmdir:   "/Library/Java/JavaVirtualMachines/jdk-#{version.before_comma}.jdk"
+
+  caveats do
+    license 'https://www.oracle.com/technetwork/java/javase/terms/license/javase-license.html'
```
Comment (commitay) [inline Casks/java.rb]: This caveat is still applicable.

Comment (RandomDSdevel) [reply Casks/java.rb]: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;🤦‍♂️  Right; not sure why I overlooked that this should be kept when first preparing this PR.  I'll fix it.

Comment (commitay) [reply Casks/java.rb]: I've already fixed the caveat and style errors.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Ah, looks like you beat me to it!  Thanks, @commitay!

**Pull request merged**